packages/react-dom/src/FadeIn.tsx

```diff
@@ -0,0 +1,61 @@
+import { type CSSProperties, type ComponentPropsWithoutRef, type ElementType } from 'react'
+import { InView } from './InView'
+import { type InViewOptions } from './useInView'
+import type { OmitKeyof } from './utility-types'
+
+type FadeInProps<TAs extends ElementType> = ComponentPropsWithoutRef<TAs> & {
+  /**
+   * The element type to render.
+   */
+  as?: TAs
+  /**
+   * The delay in milliseconds before the animation starts.
+   */
+  delay?: number
+  /**
+   * The duration in milliseconds of the animation.
+   */
+  duration?: number
+  /**
+   * The timing function of the animation.
+   */
+  timingFunction?: CSSProperties['animationTimingFunction']
+  /**
+   * The style of the element.
+   */
+  style?: OmitKeyof<CSSProperties, 'opacity' | 'willChange' | 'transition'>
```

Comment:
Could you verify if there's a type issue when adding opacity to the style prop of the FadeIn component? I've added it but didn't get any type errors as expected.

Reply:
I found a solution by updating the type definition:
```tsx
type FadeInProps<TAs extends ElementType> = Omit<ComponentPropsWithoutRef<TAs>, 'style'> & {
  // ...
  style?: OmitKeyof<CSSProperties, 'opacity' | 'willChange' | 'transition'>;
};
```
This properly excludes the style prop from the base props before extending it with our custom type.

Reply:
I made utility-type Overrride & use it to prevent mistake to forget omitting keyof overriding object in [324f428](https://github.com/toss/suspensive/pull/1362/commits/324f4289997a3bc62b2cef40355ee066f0a5942b)